### DISCUSSION TITLE
Recovering focus/cursor on custom menu button click

### DISCUSSION THREAD

**Discussion by FunMiles** [Milkdown/milkdown#1216] [Q&A] [answered]
I am emulating[ the code from the playground](https://github.com/Milkdown/website/blob/main/src/components/playground-editor/index.tsx) in a svelte app. I created custom buttons and attached the calls to Svelte's `on:click`.
The actions work correctly, but after the click, the focus remains on the button and not the editor. As a result, the cursor is lost, making the whole UX unpleasant.

The code I am emulating is in react and I have switched away from react quite a few years ago and do not see what is different in the react code that makes the focus return-to/stay-with the editor.
I would appreciate any help from anyone who has done this in Svelte.

**Comment by Saul-Mirone** [comment, marked as answer]
Just call `focus` on editor view, an example is here: https://github.com/Milkdown/milkdown/blob/main/packages/components/src/link-tooltip/edit/edit-view.ts#L45

**Comment by FunMiles** [comment]
> Just call `focus` on editor view, an example is here: https://github.com/Milkdown/milkdown/blob/main/packages/components/src/link-tooltip/edit/edit-view.ts#L45

Thanks. That does indeed work. It takes one more layer  to get access to the context, but not difficult.
One more note: when typing the code, I noticed that `ctx.get(editorViewCtx)` returns an object that has a `focus` method.
Calling directly `ctx.get(editorViewCtx).focus()` does the job as well. Is there any reason to use `ctx.get(editorViewCtx).dom.focus()` instead?

**Reply by Saul-Mirone** [reply]
I think they are equal.

**Comment by natsuki-engr** [comment]
You should use mousedown event with `e.preventDefault()`, not click event.
Mousedown event causes blur event, after this, click event is called.

This is a good example to understand these event behaviors.
https://codepen.io/mudassir0909/pen/qBjvzL

**Reply by FunMiles** [reply]
That's an interesting alternative. I can see some advantages to it, not the least that it really leaves the focus/selection/etc intact.
Does it have any accessibility impact?

**Reply by FunMiles** [reply]
@natsuki-engr : After thinking about it more, using a mousedown event with `e.preventDefault()` impedes the possibility of gesture started within the button, doesn't it? From my understanding, mousedown is triggered also on touch devices, so this solution is not ideal either.

**Reply by natsuki-engr** [reply]
@FunMiles
What kind of gesture? I think there is no concern about accessibility.

**Reply by FunMiles** [reply]
Navigation backward is an example. I. E. Swiping left to right.
